requests/models.py

```diff
@@ -813,7 +813,10 @@ def iter_lines(self, chunk_size=ITER_CHUNK_SIZE, decode_unicode=None, delimiter=
             #
             # If we're using `splitlines()`, we only do this if the chunk
             # ended midway through a line.
-            incomplete_line = lines[-1] and lines[-1].endswith(chunk[-1])
+            if isinstance(chunk[-1], basestring):  # Decoded string (decode_unicode=True or Py2)
+                incomplete_line = lines[-1] and lines[-1].endswith(chunk[-1])
+            else:  # Bytestring (decode_unicode=False and Py3)
+                incomplete_line = lines[-1] and lines[-1].endswith(chr(chunk[-1]).encode('utf-8'))
```

Comment:
So, to be clear, the error in this fix was assuming that we need to tolerate `str` here. The tests were in error, not the code: `iter_content` is required to always yield `bytes`. You'll find if you go through the tests and use `bytes` everywhere the problem goes away. :grin: So we don't need this.